{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "CSnBkNcarB-H"
   },
   "source": [
    "# **Tulbad:**\n",
    "\n",
    "\n",
    "*   Kollased - välja võtta\n",
    "*   Sinised - lahku lüüa \"/\"\n",
    "*   Beež - küsitav\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "id": "zAKKYYtI7tWi"
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\ProgramData\\Anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py:3071: DtypeWarning: Columns (0,38,40) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n"
     ]
    }
   ],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "from copy import *\n",
    "import numbers\n",
    "# TODO\n",
    "data = pd.read_csv(\"clinvar_conflicting.csv\")\n",
    "data1 = deepcopy(data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 326
    },
    "id": "XZt4iHtf7tsy",
    "outputId": "89929794-1ce5-4891-bfaa-d13da86a2ea6"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AF_ESP</th>\n",
       "      <th>AF_EXAC</th>\n",
       "      <th>AF_TGP</th>\n",
       "      <th>CLNDNINCL</th>\n",
       "      <th>CLNSIGINCL</th>\n",
       "      <th>CLNVC</th>\n",
       "      <th>...</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>LoFtool</th>\n",
       "      <th>CADD_PHRED</th>\n",
       "      <th>CADD_RAW</th>\n",
       "      <th>BLOSUM62</th>\n",
       "      <th>Amino_acids algne</th>\n",
       "      <th>Amino_acids asendus</th>\n",
       "      <th>Codons algne</th>\n",
       "      <th>Codons asendus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1168180</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.0771</td>\n",
       "      <td>0.10020</td>\n",
       "      <td>0.1066</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>tolerated</td>\n",
       "      <td>benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.053</td>\n",
       "      <td>-0.208682</td>\n",
       "      <td>2.0</td>\n",
       "      <td>E</td>\n",
       "      <td>D</td>\n",
       "      <td>gaG</td>\n",
       "      <td>gaC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1470752</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>deleterious_low_confidence</td>\n",
       "      <td>benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>31.000</td>\n",
       "      <td>6.517838</td>\n",
       "      <td>-3.0</td>\n",
       "      <td>P</td>\n",
       "      <td>L</td>\n",
       "      <td>cCg</td>\n",
       "      <td>cTg</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1737942</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00001</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>deleterious</td>\n",
       "      <td>probably_damaging</td>\n",
       "      <td>NaN</td>\n",
       "      <td>28.100</td>\n",
       "      <td>6.061752</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>I</td>\n",
       "      <td>T</td>\n",
       "      <td>aTc</td>\n",
       "      <td>aCc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>2160305</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22.500</td>\n",
       "      <td>3.114491</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>S</td>\n",
       "      <td>Ggc</td>\n",
       "      <td>Agc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>2160305</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24.700</td>\n",
       "      <td>4.766224</td>\n",
       "      <td>-3.0</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Ggc</td>\n",
       "      <td>Tgc</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM      POS REF ALT  AF_ESP  AF_EXAC  AF_TGP CLNDNINCL CLNSIGINCL  \\\n",
       "0     1  1168180   G   C  0.0771  0.10020  0.1066       NaN        NaN   \n",
       "1     1  1470752   G   A  0.0000  0.00000  0.0000       NaN        NaN   \n",
       "2     1  1737942   A   G  0.0000  0.00001  0.0000       NaN        NaN   \n",
       "3     1  2160305   G   A  0.0000  0.00000  0.0000       NaN        NaN   \n",
       "4     1  2160305   G   T  0.0000  0.00000  0.0000       NaN        NaN   \n",
       "\n",
       "                       CLNVC  ...                        SIFT  \\\n",
       "0  single_nucleotide_variant  ...                   tolerated   \n",
       "1  single_nucleotide_variant  ...  deleterious_low_confidence   \n",
       "2  single_nucleotide_variant  ...                 deleterious   \n",
       "3  single_nucleotide_variant  ...                         NaN   \n",
       "4  single_nucleotide_variant  ...                         NaN   \n",
       "\n",
       "            PolyPhen LoFtool CADD_PHRED  CADD_RAW BLOSUM62 Amino_acids algne  \\\n",
       "0             benign     NaN      1.053 -0.208682      2.0                 E   \n",
       "1             benign     NaN     31.000  6.517838     -3.0                 P   \n",
       "2  probably_damaging     NaN     28.100  6.061752     -1.0                 I   \n",
       "3                NaN     NaN     22.500  3.114491      NaN                 G   \n",
       "4                NaN     NaN     24.700  4.766224     -3.0                 G   \n",
       "\n",
       "  Amino_acids asendus Codons algne Codons asendus  \n",
       "0                   D          gaG            gaC  \n",
       "1                   L          cCg            cTg  \n",
       "2                   T          aTc            aCc  \n",
       "3                   S          Ggc            Agc  \n",
       "4                   C          Ggc            Tgc  \n",
       "\n",
       "[5 rows x 32 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#muudame nende tulpade sisu\n",
    "data1[[\"Amino_acids algne\", \"Amino_acids asendus\"]] = data1.Amino_acids.str.split(\"/\",expand=True)\n",
    "data1[[\"Codons algne\", \"Codons asendus\"]] = data1.Codons.str.split(\"/\",expand=True)\n",
    "\n",
    "#eemaldame mittevajalikud tulbad\n",
    "välja = [\"Allele\", \"Amino_acids\", \"Codons\", \"CLNDN\", \"CLNDISDB\", \"CLNDISDBINCL\", \"CLNHGVS\", \"CLNVI\", \"MC\", \"SSR\", \"Feature_type\", \"Feature\", \"DISTANCE\", \"BAM_EDIT\", \"MOTIF_NAME\", \"MOTIF_POS\", \"HIGH_INF_POS\", \"MOTIF_SCORE_CHANGE\"]\n",
    "for i in välja:\n",
    "\tdata1 = data1.drop(labels=[i], axis=1)\n",
    "\n",
    "#ref tulp kõik mis on rohkem kui 1 täht\n",
    "data1.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "TCLY_ELe3Bnj",
    "outputId": "da9ec3bd-aeef-454a-880d-54af7d853b06"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "      CHROM       POS REF ALT  AF_ESP  AF_EXAC  AF_TGP CLNDNINCL CLNSIGINCL  \\\n",
      "5227     19   1397399   C   T  0.0002  0.00028  0.0000       NaN        NaN   \n",
      "5228     19   1397443   G   A  0.0712  0.07554  0.0365       NaN        NaN   \n",
      "5473     19  35530029   G   A  0.0002  0.00007  0.0000       NaN        NaN   \n",
      "5474     19  35530580   G   A  0.0003  0.00050  0.0004       NaN        NaN   \n",
      "6940      1  40539727   C   T  0.0000  0.00016  0.0000       NaN        NaN   \n",
      "...     ...       ...  ..  ..     ...      ...     ...       ...        ...   \n",
      "63480    MT     15470   T   C  0.0000  0.00000  0.0000       NaN        NaN   \n",
      "63481    MT     15511   T   C  0.0000  0.00000  0.0000       NaN        NaN   \n",
      "63482    MT     15923   A   G  0.0000  0.00000  0.0000       NaN        NaN   \n",
      "64248     X  55057617   G   C  0.0000  0.00000  0.0027       NaN        NaN   \n",
      "64358     X  73641251   A   T  0.0009  0.00144  0.0000       NaN        NaN   \n",
      "\n",
      "                           CLNVC  ORIGIN  CLASS              Consequence  \\\n",
      "5227   single_nucleotide_variant       1      1  downstream_gene_variant   \n",
      "5228   single_nucleotide_variant       1      0  downstream_gene_variant   \n",
      "5473   single_nucleotide_variant       1      1    upstream_gene_variant   \n",
      "5474   single_nucleotide_variant       1      1    upstream_gene_variant   \n",
      "6940   single_nucleotide_variant       1      1  downstream_gene_variant   \n",
      "...                          ...     ...    ...                      ...   \n",
      "63480  single_nucleotide_variant       1      1       intergenic_variant   \n",
      "63481  single_nucleotide_variant       1      1       intergenic_variant   \n",
      "63482  single_nucleotide_variant       1      1       intergenic_variant   \n",
      "64248  single_nucleotide_variant       1      1    upstream_gene_variant   \n",
      "64358  single_nucleotide_variant       1      0    upstream_gene_variant   \n",
      "\n",
      "         IMPACT   SYMBOL         BIOTYPE   EXON INTRON cDNA_position  \\\n",
      "5227   MODIFIER   NDUFS7  protein_coding  TÕRGE  TÕRGE           NaN   \n",
      "5228   MODIFIER   NDUFS7  protein_coding  TÕRGE  TÕRGE           NaN   \n",
      "5473   MODIFIER      HPN  protein_coding  TÕRGE  TÕRGE           NaN   \n",
      "5474   MODIFIER      HPN  protein_coding  TÕRGE  TÕRGE           NaN   \n",
      "6940   MODIFIER     CAP1  protein_coding  TÕRGE  TÕRGE           NaN   \n",
      "...         ...      ...             ...    ...    ...           ...   \n",
      "63480  MODIFIER      NaN             NaN  TÕRGE  TÕRGE           NaN   \n",
      "63481  MODIFIER      NaN             NaN  TÕRGE  TÕRGE           NaN   \n",
      "63482  MODIFIER      NaN             NaN  TÕRGE  TÕRGE           NaN   \n",
      "64248  MODIFIER    ALAS2  protein_coding  TÕRGE  TÕRGE           NaN   \n",
      "64358  MODIFIER  SLC16A2  protein_coding  TÕRGE  TÕRGE           NaN   \n",
      "\n",
      "      CDS_position Protein_position  STRAND SIFT PolyPhen  LoFtool  \\\n",
      "5227           NaN              NaN     1.0  NaN      NaN    0.127   \n",
      "5228           NaN              NaN     1.0  NaN      NaN    0.127   \n",
      "5473           NaN              NaN     1.0  NaN      NaN      NaN   \n",
      "5474           NaN              NaN     1.0  NaN      NaN      NaN   \n",
      "6940           NaN              NaN     1.0  NaN      NaN      NaN   \n",
      "...            ...              ...     ...  ...      ...      ...   \n",
      "63480          NaN              NaN     NaN  NaN      NaN      NaN   \n",
      "63481          NaN              NaN     NaN  NaN      NaN      NaN   \n",
      "63482          NaN              NaN     NaN  NaN      NaN      NaN   \n",
      "64248          NaN              NaN    -1.0  NaN      NaN      NaN   \n",
      "64358          NaN              NaN     1.0  NaN      NaN      NaN   \n",
      "\n",
      "       CADD_PHRED  CADD_RAW  BLOSUM62 Amino_acids algne Amino_acids asendus  \\\n",
      "5227       22.700  3.206002       NaN               NaN                 NaN   \n",
      "5228       10.240  0.927789       NaN               NaN                 NaN   \n",
      "5473       16.290  2.012176       NaN               NaN                 NaN   \n",
      "5474       27.800  5.991076       NaN               NaN                 NaN   \n",
      "6940        0.247 -0.479761       NaN               NaN                 NaN   \n",
      "...           ...       ...       ...               ...                 ...   \n",
      "63480       2.860  0.027997       NaN               NaN                 NaN   \n",
      "63481       2.081 -0.055831       NaN               NaN                 NaN   \n",
      "63482       9.734  0.844303       NaN               NaN                 NaN   \n",
      "64248       1.927 -0.074507       NaN               NaN                 NaN   \n",
      "64358      12.800  1.402671       NaN               NaN                 NaN   \n",
      "\n",
      "      Codons algne Codons asendus  \n",
      "5227           NaN            NaN  \n",
      "5228           NaN            NaN  \n",
      "5473           NaN            NaN  \n",
      "5474           NaN            NaN  \n",
      "6940           NaN            NaN  \n",
      "...            ...            ...  \n",
      "63480          NaN            NaN  \n",
      "63481          NaN            NaN  \n",
      "63482          NaN            NaN  \n",
      "64248          NaN            NaN  \n",
      "64358          NaN            NaN  \n",
      "\n",
      "[133 rows x 32 columns]\n"
     ]
    }
   ],
   "source": [
    "pd.set_option('display.max_columns', None)\n",
    "data1[\"EXON\"]=data1[\"EXON\"].astype(str)\n",
    "data1[\"INTRON\"]=data1[\"INTRON\"].astype(str)\n",
    "\n",
    "data1.loc[(data1[\"EXON\"]!='nan') & (data1[\"INTRON\"]!='nan') & (data1[\"REF\"].str.len() == 1), [\"INTRON\", \"EXON\"]] = \"TÕRGE\"\n",
    "data1.loc[data1[\"EXON\"] == data1[\"INTRON\"], [\"INTRON\", \"EXON\"]] = \"TÕRGE\"\n",
    "\n",
    "\n",
    "print(data1[data1[\"INTRON\"] == \"TÕRGE\"])\n",
    "\n",
    "\n",
    "#vaja on, et NaN'idest tekiksid eraldi tulbad mille saaks droppida\n",
    "#vaja on midagi exonite ja intronitega teha: kui on variant eksonism ei saa ta intronis enam olla ja vastupidi\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "xnCYSGUz7gjk",
    "outputId": "83d32610-b837-4913-9c32-dc24f4669941"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CHROM\n",
      "POS\n",
      "REF\n",
      "ALT\n",
      "AF_ESP\n",
      "AF_EXAC\n",
      "AF_TGP\n",
      "CLNDISDB\n",
      "CLNDISDBINCL\n",
      "CLNDN\n",
      "CLNDNINCL\n",
      "CLNHGVS\n",
      "CLNSIGINCL\n",
      "CLNVC\n",
      "CLNVI\n",
      "MC\n",
      "ORIGIN\n",
      "SSR\n",
      "CLASS\n",
      "Allele\n",
      "Consequence\n",
      "IMPACT\n",
      "SYMBOL\n",
      "Feature_type\n",
      "Feature\n",
      "BIOTYPE\n",
      "EXON\n",
      "INTRON\n",
      "cDNA_position\n",
      "CDS_position\n",
      "Protein_position\n",
      "Amino_acids\n",
      "Codons\n",
      "DISTANCE\n",
      "STRAND\n",
      "BAM_EDIT\n",
      "SIFT\n",
      "PolyPhen\n",
      "MOTIF_NAME\n",
      "MOTIF_POS\n",
      "HIGH_INF_POS\n",
      "MOTIF_SCORE_CHANGE\n",
      "LoFtool\n",
      "CADD_PHRED\n",
      "CADD_RAW\n",
      "BLOSUM62\n"
     ]
    }
   ],
   "source": [
    "#LIHTSASTI DEFINEERITAVAD TULBAD\n",
    "inttüüpi = [\"POS\", \"ORIGIN\", \"CLASS\", \"cDNA_position\", \"CDS_position\", \"Protein_position\", \"STRAND\", \"BLOSUM62\" ]\n",
    "floattüüpi = [\"AF_ESP\", \"AF_EXAC\", \"AF_TGP\", \"LoFtool\", \"CADD_PHRED\", \"CADD_RAW\" ]\n",
    "strtüüpi = [\"REF\", \"ALT\", \"Consequence\", \"IMPACT\", \"SYMBOL\", \"BIOTYPE\" , \"CLNVC\", \"IMPACT\", \"Feature_type\", \"SIFT\", \"PolyPhen\", \"Amino_acids algne\", \"Amino_acids asendus\", \"Codons algne\", \"Codons asendus\"]\n",
    "\n",
    "\n",
    "def datatypes(inttüüpi, strtüüpi, floattüüpi):\n",
    "  for i, item in inttüüpi:\n",
    "      data1[inttüüpi[i]]=data1[inttüüpi[i]].astype(int)\n",
    "\n",
    "  for i, item in strtüüpi:\n",
    "      data1[strtüüpi[i]]=data1[strtüüpi[i]].astype(str)\n",
    "  \n",
    "  for i, item in floattüüpi:\n",
    "      data1[strtüüpi[i]]=data1[strtüüpi[i]].astype(float)\n",
    "      \n",
    "\n",
    "#ISIKLIKKU LÄHENEMIST VAJAVAD TULBAD\n",
    "##CHROM\n",
    "data1[\"CHROM\"]=data1[\"CHROM\"].astype(str)\n",
    "data1.loc[data1[\"CHROM\"] == \"X\", \"CHROM\"] = \"23\"\n",
    "data1.loc[data1[\"CHROM\"] == \"MT\", \"CHROM\"] = \"24\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "colab": {
   "collapsed_sections": [],
   "name": "Clinvar projekt",
   "provenance": []
  },
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
