{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "name": "Clinvar projekt",
      "provenance": [],
      "collapsed_sections": []
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "CSnBkNcarB-H"
      },
      "source": [
        "# **Tulbad:**\n",
        "\n",
        "\n",
        "*   Kollased - välja võtta\n",
        "*   Sinised - lahku lüüa \"/\"\n",
        "*   Beež - küsitav\n",
        "\n",
        "\n",
        "\n",
        "\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "zAKKYYtI7tWi"
      },
      "source": [
        "import numpy as np\n",
        "import pandas as pd\n",
        "from copy import *\n",
        "import numbers\n",
        "# TODO\n",
        "data = pd.read_csv(\"clinvar_conflicting.csv\")\n",
        "data1 = deepcopy(data)"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "XZt4iHtf7tsy",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 326
        },
        "outputId": "89929794-1ce5-4891-bfaa-d13da86a2ea6"
      },
      "source": [
        "#muudame nende tulpade sisu\n",
        "data1[[\"Amino_acids algne\", \"Amino_acids asendus\"]] = data1.Amino_acids.str.split(\"/\",expand=True)\n",
        "data1[[\"Codons algne\", \"Codons asendus\"]] = data1.Codons.str.split(\"/\",expand=True)\n",
        "\n",
        "#eemaldame mittevajalikud tulbad\n",
        "välja = [\"Allele\", \"Amino_acids\", \"Codons\", \"CLNDN\", \"CLNDISDB\", \"CLNDISDBINCL\", \"CLNHGVS\", \"CLNVI\", \"MC\", \"SSR\", \"Feature_type\", \"Feature\", \"DISTANCE\", \"BAM_EDIT\", \"MOTIF_NAME\", \"MOTIF_POS\", \"HIGH_INF_POS\", \"MOTIF_SCORE_CHANGE\"]\n",
        "for i in välja:\n",
        "\tdata1 = data1.drop(labels=[i], axis=1)\n",
        "\n",
        "\n",
        "data1.head()"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>CHROM</th>\n",
              "      <th>POS</th>\n",
              "      <th>REF</th>\n",
              "      <th>ALT</th>\n",
              "      <th>AF_ESP</th>\n",
              "      <th>AF_EXAC</th>\n",
              "      <th>AF_TGP</th>\n",
              "      <th>CLNDNINCL</th>\n",
              "      <th>CLNSIGINCL</th>\n",
              "      <th>CLNVC</th>\n",
              "      <th>ORIGIN</th>\n",
              "      <th>CLASS</th>\n",
              "      <th>Consequence</th>\n",
              "      <th>IMPACT</th>\n",
              "      <th>SYMBOL</th>\n",
              "      <th>BIOTYPE</th>\n",
              "      <th>EXON</th>\n",
              "      <th>INTRON</th>\n",
              "      <th>cDNA_position</th>\n",
              "      <th>CDS_position</th>\n",
              "      <th>Protein_position</th>\n",
              "      <th>STRAND</th>\n",
              "      <th>SIFT</th>\n",
              "      <th>PolyPhen</th>\n",
              "      <th>LoFtool</th>\n",
              "      <th>CADD_PHRED</th>\n",
              "      <th>CADD_RAW</th>\n",
              "      <th>BLOSUM62</th>\n",
              "      <th>Amino_acids algne</th>\n",
              "      <th>Amino_acids asendus</th>\n",
              "      <th>Codons algne</th>\n",
              "      <th>Codons asendus</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>1</td>\n",
              "      <td>1168180</td>\n",
              "      <td>G</td>\n",
              "      <td>C</td>\n",
              "      <td>0.0771</td>\n",
              "      <td>0.10020</td>\n",
              "      <td>0.1066</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "      <td>missense_variant</td>\n",
              "      <td>MODERATE</td>\n",
              "      <td>B3GALT6</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>01-Jan</td>\n",
              "      <td>NaN</td>\n",
              "      <td>552.0</td>\n",
              "      <td>522.0</td>\n",
              "      <td>174.0</td>\n",
              "      <td>1.0</td>\n",
              "      <td>tolerated</td>\n",
              "      <td>benign</td>\n",
              "      <td>NaN</td>\n",
              "      <td>1.053</td>\n",
              "      <td>-0.208682</td>\n",
              "      <td>2.0</td>\n",
              "      <td>E</td>\n",
              "      <td>D</td>\n",
              "      <td>gaG</td>\n",
              "      <td>gaC</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>1</td>\n",
              "      <td>1470752</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>0.0000</td>\n",
              "      <td>0.00000</td>\n",
              "      <td>0.0000</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "      <td>missense_variant</td>\n",
              "      <td>MODERATE</td>\n",
              "      <td>TMEM240</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>04-Apr</td>\n",
              "      <td>NaN</td>\n",
              "      <td>523.0</td>\n",
              "      <td>509.0</td>\n",
              "      <td>170.0</td>\n",
              "      <td>-1.0</td>\n",
              "      <td>deleterious_low_confidence</td>\n",
              "      <td>benign</td>\n",
              "      <td>NaN</td>\n",
              "      <td>31.000</td>\n",
              "      <td>6.517838</td>\n",
              "      <td>-3.0</td>\n",
              "      <td>P</td>\n",
              "      <td>L</td>\n",
              "      <td>cCg</td>\n",
              "      <td>cTg</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>1</td>\n",
              "      <td>1737942</td>\n",
              "      <td>A</td>\n",
              "      <td>G</td>\n",
              "      <td>0.0000</td>\n",
              "      <td>0.00001</td>\n",
              "      <td>0.0000</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>35</td>\n",
              "      <td>1</td>\n",
              "      <td>missense_variant</td>\n",
              "      <td>MODERATE</td>\n",
              "      <td>GNB1</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>06-Dec</td>\n",
              "      <td>NaN</td>\n",
              "      <td>632.0</td>\n",
              "      <td>239.0</td>\n",
              "      <td>80.0</td>\n",
              "      <td>-1.0</td>\n",
              "      <td>deleterious</td>\n",
              "      <td>probably_damaging</td>\n",
              "      <td>NaN</td>\n",
              "      <td>28.100</td>\n",
              "      <td>6.061752</td>\n",
              "      <td>-1.0</td>\n",
              "      <td>I</td>\n",
              "      <td>T</td>\n",
              "      <td>aTc</td>\n",
              "      <td>aCc</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>1</td>\n",
              "      <td>2160305</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>0.0000</td>\n",
              "      <td>0.00000</td>\n",
              "      <td>0.0000</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>33</td>\n",
              "      <td>0</td>\n",
              "      <td>missense_variant</td>\n",
              "      <td>MODERATE</td>\n",
              "      <td>SKI</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>01-Jul</td>\n",
              "      <td>NaN</td>\n",
              "      <td>132.0</td>\n",
              "      <td>100.0</td>\n",
              "      <td>34.0</td>\n",
              "      <td>1.0</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>22.500</td>\n",
              "      <td>3.114491</td>\n",
              "      <td>NaN</td>\n",
              "      <td>G</td>\n",
              "      <td>S</td>\n",
              "      <td>Ggc</td>\n",
              "      <td>Agc</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>1</td>\n",
              "      <td>2160305</td>\n",
              "      <td>G</td>\n",
              "      <td>T</td>\n",
              "      <td>0.0000</td>\n",
              "      <td>0.00000</td>\n",
              "      <td>0.0000</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>single_nucleotide_variant</td>\n",
              "      <td>33</td>\n",
              "      <td>0</td>\n",
              "      <td>missense_variant</td>\n",
              "      <td>MODERATE</td>\n",
              "      <td>SKI</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>01-Jul</td>\n",
              "      <td>NaN</td>\n",
              "      <td>132.0</td>\n",
              "      <td>100.0</td>\n",
              "      <td>34.0</td>\n",
              "      <td>1.0</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>24.700</td>\n",
              "      <td>4.766224</td>\n",
              "      <td>-3.0</td>\n",
              "      <td>G</td>\n",
              "      <td>C</td>\n",
              "      <td>Ggc</td>\n",
              "      <td>Tgc</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "   CHROM      POS REF  ... Amino_acids asendus  Codons algne  Codons asendus\n",
              "0      1  1168180   G  ...                   D           gaG             gaC\n",
              "1      1  1470752   G  ...                   L           cCg             cTg\n",
              "2      1  1737942   A  ...                   T           aTc             aCc\n",
              "3      1  2160305   G  ...                   S           Ggc             Agc\n",
              "4      1  2160305   G  ...                   C           Ggc             Tgc\n",
              "\n",
              "[5 rows x 32 columns]"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 150
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "TCLY_ELe3Bnj",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "da9ec3bd-aeef-454a-880d-54af7d853b06"
      },
      "source": [
        "#data1.loc[data1.EXON == data1.INTRON, \"INTRON\"] = \"TÕRGE\"\n",
        "(data1[\"INTRON\"] == data1.EXON).count()\n",
        "#data1 = pd.get_dummies(data1,columns=list(data1.select_dtypes('object')))\n",
        "\n",
        "#vaja on, et NaN'idest tekiksid eraldi tulbad mille saaks droppida\n",
        "#vaja on midagi exonite ja intronitega teha: kui on variant eksonism ei saa ta intronis enam olla ja vastupidi\n",
        "\n",
        "\n"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "3705"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 151
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "xnCYSGUz7gjk",
        "outputId": "83d32610-b837-4913-9c32-dc24f4669941"
      },
      "source": [
        "for col in data.columns: \n",
        "    print(col) "
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "CHROM\n",
            "POS\n",
            "REF\n",
            "ALT\n",
            "AF_ESP\n",
            "AF_EXAC\n",
            "AF_TGP\n",
            "CLNDISDB\n",
            "CLNDISDBINCL\n",
            "CLNDN\n",
            "CLNDNINCL\n",
            "CLNHGVS\n",
            "CLNSIGINCL\n",
            "CLNVC\n",
            "CLNVI\n",
            "MC\n",
            "ORIGIN\n",
            "SSR\n",
            "CLASS\n",
            "Allele\n",
            "Consequence\n",
            "IMPACT\n",
            "SYMBOL\n",
            "Feature_type\n",
            "Feature\n",
            "BIOTYPE\n",
            "EXON\n",
            "INTRON\n",
            "cDNA_position\n",
            "CDS_position\n",
            "Protein_position\n",
            "Amino_acids\n",
            "Codons\n",
            "DISTANCE\n",
            "STRAND\n",
            "BAM_EDIT\n",
            "SIFT\n",
            "PolyPhen\n",
            "MOTIF_NAME\n",
            "MOTIF_POS\n",
            "HIGH_INF_POS\n",
            "MOTIF_SCORE_CHANGE\n",
            "LoFtool\n",
            "CADD_PHRED\n",
            "CADD_RAW\n",
            "BLOSUM62\n"
          ],
          "name": "stdout"
        }
      ]
    }
  ]
}