{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\Ajangu\\anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py:3071: DtypeWarning: Columns (0,38,40) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AF_ESP</th>\n",
       "      <th>AF_EXAC</th>\n",
       "      <th>AF_TGP</th>\n",
       "      <th>CLNDISDB</th>\n",
       "      <th>CLNDISDBINCL</th>\n",
       "      <th>CLNDN</th>\n",
       "      <th>CLNDNINCL</th>\n",
       "      <th>CLNHGVS</th>\n",
       "      <th>CLNSIGINCL</th>\n",
       "      <th>CLNVC</th>\n",
       "      <th>CLNVI</th>\n",
       "      <th>MC</th>\n",
       "      <th>ORIGIN</th>\n",
       "      <th>SSR</th>\n",
       "      <th>CLASS</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>BAM_EDIT</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>MOTIF_NAME</th>\n",
       "      <th>MOTIF_POS</th>\n",
       "      <th>HIGH_INF_POS</th>\n",
       "      <th>MOTIF_SCORE_CHANGE</th>\n",
       "      <th>LoFtool</th>\n",
       "      <th>CADD_PHRED</th>\n",
       "      <th>CADD_RAW</th>\n",
       "      <th>BLOSUM62</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1168180</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.0771</td>\n",
       "      <td>0.10020</td>\n",
       "      <td>0.1066</td>\n",
       "      <td>MedGen:CN169374</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_specified</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000001.10:g.1168180G&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>UniProtKB_(protein):Q96L58#VAR_059317</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>B3GALT6</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>NM_080605.3</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1/1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>552</td>\n",
       "      <td>522</td>\n",
       "      <td>174</td>\n",
       "      <td>E/D</td>\n",
       "      <td>gaG/gaC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tolerated</td>\n",
       "      <td>benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.053</td>\n",
       "      <td>-0.208682</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1470752</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>MedGen:C1843891,OMIM:607454,Orphanet:ORPHA9877...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Spinocerebellar_ataxia_21|not_provided</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000001.10:g.1470752G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>OMIM_Allelic_Variant:616101.0001|UniProtKB_(pr...</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>TMEM240</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>NM_001114748.1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>4/4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>523</td>\n",
       "      <td>509</td>\n",
       "      <td>170</td>\n",
       "      <td>P/L</td>\n",
       "      <td>cCg/cTg</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>OK</td>\n",
       "      <td>deleterious_low_confidence</td>\n",
       "      <td>benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>31.000</td>\n",
       "      <td>6.517838</td>\n",
       "      <td>-3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1737942</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00001</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>Human_Phenotype_Ontology:HP:0000486,MedGen:C00...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Strabismus|Nystagmus|Hypothyroidism|Intellectu...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000001.10:g.1737942A&gt;G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>OMIM_Allelic_Variant:139380.0002|UniProtKB_(pr...</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001623|5_prime...</td>\n",
       "      <td>35</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>GNB1</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>NM_002074.4</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>6/12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>632</td>\n",
       "      <td>239</td>\n",
       "      <td>80</td>\n",
       "      <td>I/T</td>\n",
       "      <td>aTc/aCc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>OK</td>\n",
       "      <td>deleterious</td>\n",
       "      <td>probably_damaging</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>28.100</td>\n",
       "      <td>6.061752</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>2160305</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>MedGen:C1321551,OMIM:182212,SNOMED_CT:83092002...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Shprintzen-Goldberg_syndrome|not_provided</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000001.10:g.2160305G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>OMIM_Allelic_Variant:164780.0004|UniProtKB_(pr...</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>SKI</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>XM_005244775.1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1/7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>132</td>\n",
       "      <td>100</td>\n",
       "      <td>34</td>\n",
       "      <td>G/S</td>\n",
       "      <td>Ggc/Agc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22.500</td>\n",
       "      <td>3.114491</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>2160305</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>MedGen:C1321551,OMIM:182212,SNOMED_CT:83092002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Shprintzen-Goldberg_syndrome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000001.10:g.2160305G&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>OMIM_Allelic_Variant:164780.0005|UniProtKB_(pr...</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>33</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>SKI</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>XM_005244775.1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1/7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>132</td>\n",
       "      <td>100</td>\n",
       "      <td>34</td>\n",
       "      <td>G/C</td>\n",
       "      <td>Ggc/Tgc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24.700</td>\n",
       "      <td>4.766224</td>\n",
       "      <td>-3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65183</th>\n",
       "      <td>X</td>\n",
       "      <td>154158201</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.0801</td>\n",
       "      <td>0.13923</td>\n",
       "      <td>0.1605</td>\n",
       "      <td>MedGen:C0019069,OMIM:306700,SNOMED_CT:28293008...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hereditary_factor_VIII_deficiency_disease|not_...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000023.10:g.154158201T&gt;G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>ARUP_Laboratories,_Molecular_Genetics_and_Geno...</td>\n",
       "      <td>SO:0001819|synonymous_variant</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>F8</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>NM_000132.3</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>14/26</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4035</td>\n",
       "      <td>3864</td>\n",
       "      <td>1288</td>\n",
       "      <td>S</td>\n",
       "      <td>tcA/tcC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>OK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.00158</td>\n",
       "      <td>0.105</td>\n",
       "      <td>-0.630908</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65184</th>\n",
       "      <td>X</td>\n",
       "      <td>154159118</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.0020</td>\n",
       "      <td>0.00060</td>\n",
       "      <td>0.0013</td>\n",
       "      <td>MedGen:CN169374|MedGen:CN239152</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_specified|Hemophilia_A,_FVIII_Deficiency</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000023.10:g.154159118C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>ARUP_Laboratories,_Molecular_Genetics_and_Geno...</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>F8</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>NM_000132.3</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>14/26</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3118</td>\n",
       "      <td>2947</td>\n",
       "      <td>983</td>\n",
       "      <td>V/I</td>\n",
       "      <td>Gta/Ata</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>OK</td>\n",
       "      <td>tolerated</td>\n",
       "      <td>benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.00158</td>\n",
       "      <td>0.002</td>\n",
       "      <td>-1.731470</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65185</th>\n",
       "      <td>X</td>\n",
       "      <td>154194886</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.0125</td>\n",
       "      <td>0.00370</td>\n",
       "      <td>0.0111</td>\n",
       "      <td>MedGen:CN169374|MedGen:CN239152</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_specified|Hemophilia_A,_FVIII_Deficiency</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000023.10:g.154194886C&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>ARUP_Laboratories,_Molecular_Genetics_and_Geno...</td>\n",
       "      <td>SO:0001819|synonymous_variant</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>T</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>F8</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>NM_000132.3</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>8/26</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1257</td>\n",
       "      <td>1086</td>\n",
       "      <td>362</td>\n",
       "      <td>A</td>\n",
       "      <td>gcG/gcA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>OK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.00158</td>\n",
       "      <td>12.850</td>\n",
       "      <td>1.412434</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65186</th>\n",
       "      <td>X</td>\n",
       "      <td>154490187</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.0003</td>\n",
       "      <td>0.00034</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>MedGen:C3501611,Orphanet:ORPHA777|MedGen:CN169374</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Non-syndromic_X-linked_intellectual_disability...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000023.10:g.154490187T&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>Illumina_Clinical_Services_Laboratory,Illumina...</td>\n",
       "      <td>SO:0001819|synonymous_variant</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>C</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>RAB39B</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>NM_171998.2</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>2/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>822</td>\n",
       "      <td>543</td>\n",
       "      <td>181</td>\n",
       "      <td>T</td>\n",
       "      <td>acA/acG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.130</td>\n",
       "      <td>-0.592415</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65187</th>\n",
       "      <td>X</td>\n",
       "      <td>154508542</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.0019</td>\n",
       "      <td>0.00267</td>\n",
       "      <td>0.0008</td>\n",
       "      <td>MedGen:CN169374|MedGen:CN517202</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not_specified|not_provided</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NC_000023.10:g.154508542G&gt;C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>C</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>CLIC2</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>XM_005274646.1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>6/7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>791</td>\n",
       "      <td>532</td>\n",
       "      <td>178</td>\n",
       "      <td>P/A</td>\n",
       "      <td>Cca/Gca</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.14000</td>\n",
       "      <td>0.046</td>\n",
       "      <td>-0.786513</td>\n",
       "      <td>-1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>65188 rows × 46 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS REF ALT  AF_ESP  AF_EXAC  AF_TGP  \\\n",
       "0         1    1168180   G   C  0.0771  0.10020  0.1066   \n",
       "1         1    1470752   G   A  0.0000  0.00000  0.0000   \n",
       "2         1    1737942   A   G  0.0000  0.00001  0.0000   \n",
       "3         1    2160305   G   A  0.0000  0.00000  0.0000   \n",
       "4         1    2160305   G   T  0.0000  0.00000  0.0000   \n",
       "...     ...        ...  ..  ..     ...      ...     ...   \n",
       "65183     X  154158201   T   G  0.0801  0.13923  0.1605   \n",
       "65184     X  154159118   C   T  0.0020  0.00060  0.0013   \n",
       "65185     X  154194886   C   T  0.0125  0.00370  0.0111   \n",
       "65186     X  154490187   T   C  0.0003  0.00034  0.0000   \n",
       "65187     X  154508542   G   C  0.0019  0.00267  0.0008   \n",
       "\n",
       "                                                CLNDISDB CLNDISDBINCL  \\\n",
       "0                                        MedGen:CN169374          NaN   \n",
       "1      MedGen:C1843891,OMIM:607454,Orphanet:ORPHA9877...          NaN   \n",
       "2      Human_Phenotype_Ontology:HP:0000486,MedGen:C00...          NaN   \n",
       "3      MedGen:C1321551,OMIM:182212,SNOMED_CT:83092002...          NaN   \n",
       "4         MedGen:C1321551,OMIM:182212,SNOMED_CT:83092002          NaN   \n",
       "...                                                  ...          ...   \n",
       "65183  MedGen:C0019069,OMIM:306700,SNOMED_CT:28293008...          NaN   \n",
       "65184                    MedGen:CN169374|MedGen:CN239152          NaN   \n",
       "65185                    MedGen:CN169374|MedGen:CN239152          NaN   \n",
       "65186  MedGen:C3501611,Orphanet:ORPHA777|MedGen:CN169374          NaN   \n",
       "65187                    MedGen:CN169374|MedGen:CN517202          NaN   \n",
       "\n",
       "                                                   CLNDN CLNDNINCL  \\\n",
       "0                                          not_specified       NaN   \n",
       "1                 Spinocerebellar_ataxia_21|not_provided       NaN   \n",
       "2      Strabismus|Nystagmus|Hypothyroidism|Intellectu...       NaN   \n",
       "3              Shprintzen-Goldberg_syndrome|not_provided       NaN   \n",
       "4                           Shprintzen-Goldberg_syndrome       NaN   \n",
       "...                                                  ...       ...   \n",
       "65183  Hereditary_factor_VIII_deficiency_disease|not_...       NaN   \n",
       "65184       not_specified|Hemophilia_A,_FVIII_Deficiency       NaN   \n",
       "65185       not_specified|Hemophilia_A,_FVIII_Deficiency       NaN   \n",
       "65186  Non-syndromic_X-linked_intellectual_disability...       NaN   \n",
       "65187                         not_specified|not_provided       NaN   \n",
       "\n",
       "                           CLNHGVS CLNSIGINCL                      CLNVC  \\\n",
       "0        NC_000001.10:g.1168180G>C        NaN  single_nucleotide_variant   \n",
       "1        NC_000001.10:g.1470752G>A        NaN  single_nucleotide_variant   \n",
       "2        NC_000001.10:g.1737942A>G        NaN  single_nucleotide_variant   \n",
       "3        NC_000001.10:g.2160305G>A        NaN  single_nucleotide_variant   \n",
       "4        NC_000001.10:g.2160305G>T        NaN  single_nucleotide_variant   \n",
       "...                            ...        ...                        ...   \n",
       "65183  NC_000023.10:g.154158201T>G        NaN  single_nucleotide_variant   \n",
       "65184  NC_000023.10:g.154159118C>T        NaN  single_nucleotide_variant   \n",
       "65185  NC_000023.10:g.154194886C>T        NaN  single_nucleotide_variant   \n",
       "65186  NC_000023.10:g.154490187T>C        NaN  single_nucleotide_variant   \n",
       "65187  NC_000023.10:g.154508542G>C        NaN  single_nucleotide_variant   \n",
       "\n",
       "                                                   CLNVI  \\\n",
       "0                  UniProtKB_(protein):Q96L58#VAR_059317   \n",
       "1      OMIM_Allelic_Variant:616101.0001|UniProtKB_(pr...   \n",
       "2      OMIM_Allelic_Variant:139380.0002|UniProtKB_(pr...   \n",
       "3      OMIM_Allelic_Variant:164780.0004|UniProtKB_(pr...   \n",
       "4      OMIM_Allelic_Variant:164780.0005|UniProtKB_(pr...   \n",
       "...                                                  ...   \n",
       "65183  ARUP_Laboratories,_Molecular_Genetics_and_Geno...   \n",
       "65184  ARUP_Laboratories,_Molecular_Genetics_and_Geno...   \n",
       "65185  ARUP_Laboratories,_Molecular_Genetics_and_Geno...   \n",
       "65186  Illumina_Clinical_Services_Laboratory,Illumina...   \n",
       "65187                                                NaN   \n",
       "\n",
       "                                                      MC  ORIGIN  SSR  CLASS  \\\n",
       "0                            SO:0001583|missense_variant       1  NaN      0   \n",
       "1                            SO:0001583|missense_variant       1  NaN      0   \n",
       "2      SO:0001583|missense_variant,SO:0001623|5_prime...      35  NaN      1   \n",
       "3                            SO:0001583|missense_variant      33  NaN      0   \n",
       "4                            SO:0001583|missense_variant      33  NaN      0   \n",
       "...                                                  ...     ...  ...    ...   \n",
       "65183                      SO:0001819|synonymous_variant       1  NaN      0   \n",
       "65184                        SO:0001583|missense_variant       1  NaN      1   \n",
       "65185                      SO:0001819|synonymous_variant       1  NaN      0   \n",
       "65186                      SO:0001819|synonymous_variant       1  NaN      0   \n",
       "65187                        SO:0001583|missense_variant       1  NaN      0   \n",
       "\n",
       "      Allele         Consequence    IMPACT   SYMBOL Feature_type  \\\n",
       "0          C    missense_variant  MODERATE  B3GALT6   Transcript   \n",
       "1          A    missense_variant  MODERATE  TMEM240   Transcript   \n",
       "2          G    missense_variant  MODERATE     GNB1   Transcript   \n",
       "3          A    missense_variant  MODERATE      SKI   Transcript   \n",
       "4          T    missense_variant  MODERATE      SKI   Transcript   \n",
       "...      ...                 ...       ...      ...          ...   \n",
       "65183      G  synonymous_variant       LOW       F8   Transcript   \n",
       "65184      T    missense_variant  MODERATE       F8   Transcript   \n",
       "65185      T  synonymous_variant       LOW       F8   Transcript   \n",
       "65186      C  synonymous_variant       LOW   RAB39B   Transcript   \n",
       "65187      C    missense_variant  MODERATE    CLIC2   Transcript   \n",
       "\n",
       "              Feature         BIOTYPE   EXON INTRON cDNA_position  \\\n",
       "0         NM_080605.3  protein_coding    1/1    NaN           552   \n",
       "1      NM_001114748.1  protein_coding    4/4    NaN           523   \n",
       "2         NM_002074.4  protein_coding   6/12    NaN           632   \n",
       "3      XM_005244775.1  protein_coding    1/7    NaN           132   \n",
       "4      XM_005244775.1  protein_coding    1/7    NaN           132   \n",
       "...               ...             ...    ...    ...           ...   \n",
       "65183     NM_000132.3  protein_coding  14/26    NaN          4035   \n",
       "65184     NM_000132.3  protein_coding  14/26    NaN          3118   \n",
       "65185     NM_000132.3  protein_coding   8/26    NaN          1257   \n",
       "65186     NM_171998.2  protein_coding    2/2    NaN           822   \n",
       "65187  XM_005274646.1  protein_coding    6/7    NaN           791   \n",
       "\n",
       "      CDS_position Protein_position Amino_acids   Codons  DISTANCE  STRAND  \\\n",
       "0              522              174         E/D  gaG/gaC       NaN     1.0   \n",
       "1              509              170         P/L  cCg/cTg       NaN    -1.0   \n",
       "2              239               80         I/T  aTc/aCc       NaN    -1.0   \n",
       "3              100               34         G/S  Ggc/Agc       NaN     1.0   \n",
       "4              100               34         G/C  Ggc/Tgc       NaN     1.0   \n",
       "...            ...              ...         ...      ...       ...     ...   \n",
       "65183         3864             1288           S  tcA/tcC       NaN    -1.0   \n",
       "65184         2947              983         V/I  Gta/Ata       NaN    -1.0   \n",
       "65185         1086              362           A  gcG/gcA       NaN    -1.0   \n",
       "65186          543              181           T  acA/acG       NaN    -1.0   \n",
       "65187          532              178         P/A  Cca/Gca       NaN    -1.0   \n",
       "\n",
       "      BAM_EDIT                        SIFT           PolyPhen MOTIF_NAME  \\\n",
       "0          NaN                   tolerated             benign        NaN   \n",
       "1           OK  deleterious_low_confidence             benign        NaN   \n",
       "2           OK                 deleterious  probably_damaging        NaN   \n",
       "3          NaN                         NaN                NaN        NaN   \n",
       "4          NaN                         NaN                NaN        NaN   \n",
       "...        ...                         ...                ...        ...   \n",
       "65183       OK                         NaN                NaN        NaN   \n",
       "65184       OK                   tolerated             benign        NaN   \n",
       "65185       OK                         NaN                NaN        NaN   \n",
       "65186      NaN                         NaN                NaN        NaN   \n",
       "65187      NaN                         NaN                NaN        NaN   \n",
       "\n",
       "       MOTIF_POS HIGH_INF_POS  MOTIF_SCORE_CHANGE  LoFtool  CADD_PHRED  \\\n",
       "0            NaN          NaN                 NaN      NaN       1.053   \n",
       "1            NaN          NaN                 NaN      NaN      31.000   \n",
       "2            NaN          NaN                 NaN      NaN      28.100   \n",
       "3            NaN          NaN                 NaN      NaN      22.500   \n",
       "4            NaN          NaN                 NaN      NaN      24.700   \n",
       "...          ...          ...                 ...      ...         ...   \n",
       "65183        NaN          NaN                 NaN  0.00158       0.105   \n",
       "65184        NaN          NaN                 NaN  0.00158       0.002   \n",
       "65185        NaN          NaN                 NaN  0.00158      12.850   \n",
       "65186        NaN          NaN                 NaN      NaN       0.130   \n",
       "65187        NaN          NaN                 NaN  0.14000       0.046   \n",
       "\n",
       "       CADD_RAW  BLOSUM62  \n",
       "0     -0.208682       2.0  \n",
       "1      6.517838      -3.0  \n",
       "2      6.061752      -1.0  \n",
       "3      3.114491       NaN  \n",
       "4      4.766224      -3.0  \n",
       "...         ...       ...  \n",
       "65183 -0.630908       NaN  \n",
       "65184 -1.731470       3.0  \n",
       "65185  1.412434       NaN  \n",
       "65186 -0.592415       NaN  \n",
       "65187 -0.786513      -1.0  \n",
       "\n",
       "[65188 rows x 46 columns]"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import numbers\n",
    "from copy import *\n",
    "from sklearn.metrics import accuracy_score\n",
    "from sklearn.ensemble import RandomForestClassifier\n",
    "from sklearn.model_selection import train_test_split\n",
    "\n",
    "# TODO\n",
    "pd.set_option('display.max_columns', None)\n",
    "#data = pd.read_csv(\"clinvar_conflicting.csv\")\n",
    "data = pd.read_table(\"clinvar_conflicting_1.txt\",sep=',')\n",
    "data1 = deepcopy(data)\n",
    "\n",
    "\n",
    "data1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#ISIKLIKKU LÄHENEMIST VAJAVAD TULBAD\n",
    "\n",
    "#muudame nende tulpade sisu\n",
    "data1[[\"Amino_acids_initial\", \"Amino_acids_replacement\"]] = data1.Amino_acids.str.split(\"/\",expand=True)\n",
    "data1[[\"Codons_initial\", \"Codons_replacement\"]] = data1.Codons.str.split(\"/\",expand=True)\n",
    "\n",
    "data1[\"EXON\"]=data1[\"EXON\"].astype(str)\n",
    "data1[\"INTRON\"]=data1[\"INTRON\"].astype(str)\n",
    "\n",
    "#data1[\"EXON\"].isnull().sum()\n",
    "data1.loc[data1[\"EXON\"] == data1[\"INTRON\"], [\"INTRON\", \"EXON\"]] = np.nan\n",
    "data1.loc[(data1[\"EXON\"]!='nan') & (data1[\"INTRON\"]!='nan') & (data1[\"REF\"].str.len() == 1), [\"INTRON\", \"EXON\"]] = np.nan\n",
    "\n",
    "#viskame välja kõik nan valued\n",
    "data1 = data1[data1['EXON'].notna()]\n",
    "data1 = data1[data1['INTRON'].notna()]\n",
    "\n",
    "data1[[\"EXON_start\", \"EXON_end\"]] = data1.EXON.str.split(\"/\",expand=True)\n",
    "data1[[\"INTRON_start\", \"INTRON_end\"]] = data1.INTRON.str.split(\"/\",expand=True)\n",
    "\n",
    "\n",
    "##CHROM\n",
    "data1[\"CHROM\"]=data1[\"CHROM\"].astype(str)\n",
    "data1.loc[data1[\"CHROM\"] == \"X\", \"CHROM\"] = \"23\"\n",
    "data1.loc[data1[\"CHROM\"] == \"MT\", \"CHROM\"] = \"24\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#LIHTSASTI DEFINEERITAVAD TULBAD\n",
    "inttüüpi = [\"POS\", \"ORIGIN\", \"CLASS\", \"CHROM\"]\n",
    "floattüüpi = [\"AF_ESP\", \"AF_EXAC\", \"AF_TGP\", \"LoFtool\", \"CADD_PHRED\", \"CADD_RAW\" ,  \"STRAND\", \"BLOSUM62\", \"EXON_start\", \"EXON_end\", \"INTRON_start\", \"INTRON_end\"]\n",
    "strtüüpi = [\"REF\", \"ALT\", \"Consequence\", \"IMPACT\", \"SYMBOL\", \"BIOTYPE\" , \"CLNVC\", \"IMPACT\", \"SIFT\", \"PolyPhen\", \"Amino_acids_initial\", \"Amino_acids_replacement\", \"Codons_initial\", \"Codons_replacement\", \"cDNA_position\" , \"CDS_position\", \"Protein_position\"]\n",
    "\n",
    "\n",
    "def datatypes(inttüüpi, strtüüpi, floattüüpi):\n",
    "  for i in range(len(inttüüpi)):\n",
    "      data1[inttüüpi[i]]=data1[inttüüpi[i]].astype(int)\n",
    "\n",
    "  for i in range(len(strtüüpi)):\n",
    "      data1[strtüüpi[i]]=data1[strtüüpi[i]].astype(str)\n",
    "  \n",
    "  for i in range(len(floattüüpi)):\n",
    "      data1[floattüüpi[i]]=data1[floattüüpi[i]].astype(float).round(decimals=5)\n",
    "      \n",
    "datatypes(inttüüpi, strtüüpi, floattüüpi)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "#CDNA, CDS ja protein tulpade töötlus\n",
    "data1[[\"cDNA_pos_start\" , \"cDNA_pos_end\"]] = data1.cDNA_position.str.split(\"-\",expand=True)\n",
    "data1[[\"CDS_pos_start\" , \"CDS_pos_end\"]] = data1.CDS_position.str.split(\"-\",expand=True)\n",
    "data1[[\"Protein_pos_start\" , \"Protein_pos_end\"]] = data1.Protein_position.str.split(\"-\",expand=True)\n",
    "\n",
    "data1.loc[data1[\"cDNA_pos_end\"].isnull(), \"cDNA_pos_end\"] = data1[\"cDNA_pos_start\"]\n",
    "data1.loc[data1[\"CDS_pos_end\"].isnull(), \"CDS_pos_end\"] = data1[\"CDS_pos_start\"]\n",
    "data1.loc[data1[\"Protein_pos_end\"].isnull(), \"Protein_pos_end\"] = data1[\"Protein_pos_start\"]\n",
    "\n",
    "data1.loc[data1[\"cDNA_pos_end\"].str.contains(\"?\", regex=False, na=False), \"cDNA_pos_end\"] = np.nan\n",
    "data1.loc[data1[\"CDS_pos_end\"].str.contains(\"?\", regex=False, na=False), \"CDS_pos_end\"] = np.nan\n",
    "data1.loc[data1[\"Protein_pos_end\"].str.contains(\"?\", regex=False, na=False), \"Protein_pos_end\"] = np.nan\n",
    "\n",
    "data1.loc[data1[\"cDNA_pos_start\"].str.contains(\"?\", regex=False, na=False), \"cDNA_pos_start\"] = np.nan\n",
    "data1.loc[data1[\"CDS_pos_start\"].str.contains(\"?\", regex=False, na=False), \"CDS_pos_start\"] = np.nan\n",
    "data1.loc[data1[\"Protein_pos_start\"].str.contains(\"?\", regex=False, na=False), \"Protein_pos_start\"] = np.nan\n",
    "\n",
    "floatiks_ja_nan_välja = [\"cDNA_pos_start\" , \"cDNA_pos_end\", \"CDS_pos_start\" , \"CDS_pos_end\", \"Protein_pos_start\" , \"Protein_pos_end\"]\n",
    "for i in range(len(floatiks_ja_nan_välja)):\n",
    "  data1[floatiks_ja_nan_välja[i]]=data1[floatiks_ja_nan_välja[i]].astype(float).round(decimals=5)\n",
    "\n",
    "for i in range(len(floatiks_ja_nan_välja)):\n",
    "  data1 = data1[data1[floatiks_ja_nan_välja[i]].notna()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "välja = [\"cDNA_position\",\"CDS_position\",\"Protein_position\",\"Allele\", \"Amino_acids\", \"Codons\", \"EXON\",\"INTRON\", \"CLNDN\", \"CLNDISDB\",\"CLNDNINCL\", \"CLNSIGINCL\", \"CLNDISDBINCL\", \"CLNHGVS\", \"CLNVI\", \"MC\", \"SSR\", \"Feature_type\", \"Feature\", \"DISTANCE\", \"BAM_EDIT\", \"MOTIF_NAME\", \"MOTIF_POS\", \"HIGH_INF_POS\", \"MOTIF_SCORE_CHANGE\"]\n",
    "for i in välja:\n",
    "    data1 = data1.drop(labels=[i], axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "30622"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#asendame nendes tulpades nan väärtused tulba keskmisega\n",
    "keskmised=[\"LoFtool\",\"CADD_PHRED\",\"CADD_RAW\",\"BLOSUM62\"]\n",
    "for i in range(len(keskmised)):\n",
    "    data1.loc[data1[keskmised[i]].notna()==False, keskmised[i]] = data1[keskmised[i]].mean()\n",
    "\n",
    "#asendame nendes tulpades nan väärtused 0-ga\n",
    "for i in range(len(floattüüpi)):\n",
    "    data1.loc[data1[floattüüpi[i]].notna()==False, floattüüpi[i]] = 0\n",
    "        \n",
    "data1.AF_ESP[data1.AF_ESP == 0].count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['CHROM', 'POS', 'REF', 'ALT', 'AF_ESP', 'AF_EXAC', 'AF_TGP', 'CLNVC', 'ORIGIN', 'CLASS', 'Consequence', 'IMPACT', 'SYMBOL', 'BIOTYPE', 'STRAND', 'SIFT', 'PolyPhen', 'LoFtool', 'CADD_PHRED', 'CADD_RAW', 'BLOSUM62', 'Amino_acids_initial', 'Amino_acids_replacement', 'Codons_initial', 'Codons_replacement', 'EXON_start', 'EXON_end', 'INTRON_start', 'INTRON_end', 'cDNA_pos_start', 'cDNA_pos_end', 'CDS_pos_start', 'CDS_pos_end', 'Protein_pos_start', 'Protein_pos_end']\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "CHROM                      0\n",
       "POS                        0\n",
       "REF                        0\n",
       "ALT                        0\n",
       "AF_ESP                     0\n",
       "AF_EXAC                    0\n",
       "AF_TGP                     0\n",
       "CLNVC                      0\n",
       "ORIGIN                     0\n",
       "CLASS                      0\n",
       "Consequence                0\n",
       "IMPACT                     0\n",
       "SYMBOL                     0\n",
       "BIOTYPE                    0\n",
       "STRAND                     0\n",
       "SIFT                       0\n",
       "PolyPhen                   0\n",
       "LoFtool                    0\n",
       "CADD_PHRED                 0\n",
       "CADD_RAW                   0\n",
       "BLOSUM62                   0\n",
       "Amino_acids_initial        0\n",
       "Amino_acids_replacement    0\n",
       "Codons_initial             0\n",
       "Codons_replacement         0\n",
       "EXON_start                 0\n",
       "EXON_end                   0\n",
       "INTRON_start               0\n",
       "INTRON_end                 0\n",
       "cDNA_pos_start             0\n",
       "cDNA_pos_end               0\n",
       "CDS_pos_start              0\n",
       "CDS_pos_end                0\n",
       "Protein_pos_start          0\n",
       "Protein_pos_end            0\n",
       "dtype: int64"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mitteInt = floattüüpi + strtüüpi\n",
    "\n",
    "valja = [\"cDNA_position\" , \"CDS_position\", \"Protein_position\"]\n",
    "for i in valja:\n",
    "    mitteInt.remove(i)\n",
    "\n",
    "for i in range(len(mitteInt)):\n",
    "    if data1[mitteInt[i]].isnull().values.any():\n",
    "        print(mitteInt[i])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "#onehotencoding\n",
    "data1 = pd.get_dummies(data1,columns=[\"REF\", \"ALT\", \"CLNVC\", \"Consequence\", \"IMPACT\", \"SYMBOL\", \"BIOTYPE\",\"SIFT\",\n",
    "                                      \"PolyPhen\",\"Amino_acids_initial\", \n",
    "                                      \"Amino_acids_replacement\",\"Codons_initial\",\"Codons_replacement\"], dummy_na=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CHROM                      0\n",
       "POS                        0\n",
       "AF_ESP                     0\n",
       "AF_EXAC                    0\n",
       "AF_TGP                     0\n",
       "                          ..\n",
       "Codons_replacement_ttg     0\n",
       "Codons_replacement_ttgg    0\n",
       "Codons_replacement_ttgt    0\n",
       "Codons_replacement_ttt     0\n",
       "Codons_replacement_nan     0\n",
       "Length: 6410, dtype: int64"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tulbad = list(data1.columns)\n",
    "for i in range(len(tulbad)):\n",
    "    if data1[tulbad[i]].isnull().values.any():\n",
    "        print(tulbad[i])\n",
    "        \n",
    "data1.isnull().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "X_train, X_test, y_train, y_test = train_test_split(data1.drop(labels='CLASS',axis=1),data1.CLASS,train_size=0.7,random_state=0)\n",
    "\n",
    "\n",
    "#mudeli fittimine\n",
    "rf = RandomForestClassifier(n_estimators=100,max_depth=4,max_features=\"log2\",random_state=0).fit(X_train, y_train)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Accuracy score:\n",
      "75.09 \n"
     ]
    }
   ],
   "source": [
    "y_pred = rf.predict(X_test)\n",
    "acc_score=accuracy_score(y_pred, y_test)\n",
    "print(\"Accuracy score:\\n%s \" %round(acc_score*100,2))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
